lib/src/test/java/io/ably/lib/test/realtime/RealtimeInitTest.java

```diff
@@ -219,4 +219,62 @@ public void println(int severity, String tag, String msg, Throwable tr) {
 			if(ably != null) ably.close();
 		}
 	}
+
+	/**
+	 * Init library with 'production' environment
+	 * Spec: RTC1e
+	 */
+	@Test
+	public void init_production_environment() {
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = new ClientOptions(testVars.keys[0].keyStr);
+			opts.environment = "production";
+			AblyRealtime ably = new AblyRealtime(opts);
+			assertEquals("Unexpected host mismatch", Defaults.HOST_REALTIME, ably.options.realtimeHost);
+		} catch (AblyException e) {
+			e.printStackTrace();
+			fail("init_production_environment: Unexpected exception instantiating library");
+		}
+	}
+
+	/**
+	 * Init library with given environment
+	 * Spec: RTC1e
+	 */
+	@Test
+	public void init_given_environment() {
+		final String givenEnvironment = "staging";
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = new ClientOptions(testVars.keys[0].keyStr);
+			opts.environment = givenEnvironment;
+			AblyRealtime ably = new AblyRealtime(opts);
+			assertEquals("Unexpected host mismatch", String.format("%s-%s", givenEnvironment, Defaults.HOST_REALTIME), ably.options.realtimeHost);
+		} catch (AblyException e) {
+			e.printStackTrace();
+			fail("init_given_environment: Unexpected exception instantiating library");
+		}
+	}
+
+	/**
+	 * Init library with given environment and specified host
+	 * Spec: RTC1e
+	 */
+	@Test
+	public void init_given_host_environment() {
+		final String givenEnvironment = "staging";
+		final String specifiedHost = "fake.ably.io";
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = new ClientOptions(testVars.keys[0].keyStr);
+			opts.realtimeHost = specifiedHost;
+			opts.environment = givenEnvironment;
```

Comment:
Yes, that is why it states it is never valid to specify both because it is unnecessary i.e. either a client will have a specific host they want to connect to, or they will use our built in environment naming convention, but never both